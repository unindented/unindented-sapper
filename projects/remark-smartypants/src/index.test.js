const stringify = require("rehype-stringify");
const remark2rehype = require("remark-rehype");
const markdown = require("remark-parse");
const unified = require("unified");
const smartypants = require(".");

const process = (text, options) =>
  new Promise((resolve, reject) => {
    unified()
      .use(markdown)
      .use(smartypants, options)
      .use(remark2rehype)
      .use(stringify)
      .process(text, (err, contents) => {
        if (!err) {
          resolve(contents.contents);
        } else {
          reject(err);
        }
      });
  });

describe("remark-smartypants", () => {
  it("converts quotes to smart quotes", () => {
    const text = `
John said, "I saw Lucy at lunch, she told me 'Mary wants you to get some ice cream on your way home'. I think I will get some at Ben and Jerry's, on Gloucester Road."
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("converts two dashes to an en-dash", () => {
    const text = `
Community-associated Methicillin-Resistant *Staphylococcus aureus* in Outpatients, United States, 1999--2006
`;
    return expect(
      process(text, { dashes: "oldschool" })
    ).resolves.toMatchSnapshot();
  });

  it("converts three dashes to an em-dash", () => {
    const text = `
> Proper words in proper places make the true definition of a style.
>
> --- Jonathan Swift
`;
    return expect(
      process(text, { dashes: "oldschool" })
    ).resolves.toMatchSnapshot();
  });
});
