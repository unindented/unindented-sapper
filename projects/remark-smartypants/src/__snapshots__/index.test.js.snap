// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark-smartypants converts quotes to smart quotes 1`] = `"<p>John said, “I saw Lucy at lunch, she told me ‘Mary wants you to get some ice cream on your way home’. I think I will get some at Ben and Jerry’s, on Gloucester Road.”</p>"`;

exports[`remark-smartypants converts three dashes to an em-dash 1`] = `
"<blockquote>
<p>Proper words in proper places make the true definition of a style.</p>
<p>— Jonathan Swift</p>
</blockquote>"
`;

exports[`remark-smartypants converts two dashes to an en-dash 1`] = `"<p>Community-associated Methicillin-Resistant <em>Staphylococcus aureus</em> in Outpatients, United States, 1999–2006</p>"`;
